nnvm/python/nnvm/frontend/tensorflow.py

```diff
@@ -1137,21 +1191,30 @@ def from_tensorflow(self, graph, layout="NHWC", shape=None):
                 # Pass the target layout
                 attr["_target_layout"] = layout
 
-                #ToDo: Some of the tensorflow operators internaly maintain
-                #execution layers and its output name will the layer number along with
-                #graph node name.eg: Node name:- 'Model/RNN/cell_0/RnnCell', but the
-                #output name will be 'Model/RNN/cell_0/RnnCell:0'. In this case,
-                #the digit has to be ignored.
-                if ":" in node.input[0]:
-                    in_name, _ = node.input[0].split(':')
-                    node.input[0] = in_name
-
                 # Fill shapes for all inputs in a list
                 inputs = []
                 for i in node.input:
+                    #ToDo: Some of the tensorflow operators internaly maintain
+                    #execution layers and its output name will the layer number along with
+                    #graph node name.eg: Node name:- 'Model/RNN/cell_0/RnnCell', but the
+                    #output name will be 'Model/RNN/cell_0/RnnCell:0'. In this case,
+                    #the number has to be ignored for single-output nodes.
+                    #On the other hand, for multi-output nodes the number indicates the used output,
+                    #and the lack of the number implies 0
+                    if ":" in i:
+                        in_name, num_layer = i.split(':')
+                        i = in_name
+                        num_layer = int(num_layer)
+                    else:
+                        num_layer = 0
+
                     if i in self._nodes:
-                        inputs.append(self._nodes[i])
-                        input_shapes[self._nodes[i]] = self._output_shapes[i]
+                        tvm_n = self._nodes[i]
+                        outputs = tvm_n.list_output_names()
+                        if len(outputs) > 1:
+                            tvm_n = tvm_n[num_layer]
+                        inputs.append(tvm_n)
+                        input_shapes[tvm_n] = self._output_shapes[i]
```

Comment:
Should it be the following for multiple outputs cases?
`input_shapes[tvm_n] = (self._output_shapes[i])[num_layer]`

Reply:
Yes, I think so. I've added a commit which fixes this, but discovered a problem while trying to create a test case distinguishing them.